addons/website_sale/controllers/main.py

```diff
@@ -228,6 +236,7 @@ def shop(self, page=0, category=None, search='', **post):
             'attributes': attributes,
             'compute_currency': compute_currency,
             'keep': keep,
+            'parent_category': recursive_category(category) if category else [],
```

Comment:
Here we need to keep open all parent categories if one of its child category is active.
Now in the Qweb template parent categories are rendered before it's child category, we need to provide list of parent categories of active category. (Because we want to minimize javascript for this task, we have done most of the things from template)